intellij/src/saros/intellij/eventhandler/editor/editorstate/EditorStatusChangeActivityDispatcher.java

```diff
@@ -45,26 +55,39 @@ public EditorStatusChangeActivityDispatcher(
   /**
    * Calls {@link LocalEditorHandler#closeEditor(Project, VirtualFile)}.
    *
+   * <p>Does nothing if the closed editor is not a text editor.
+   *
    * @param virtualFile the file whose editor was closed
    * @see FileEditorManagerListener#fileClosed(FileEditorManager, VirtualFile)
    */
   private void generateEditorClosedActivity(@NotNull VirtualFile virtualFile) {
-    localEditorHandler.closeEditor(project, virtualFile);
+    if (ProjectAPI.isOpenInTextEditor(project, virtualFile)) {
+      localEditorHandler.closeEditor(project, virtualFile);
+    }
   }
 
   /**
    * Calls {@link LocalEditorHandler#activateEditor(Project, VirtualFile)}.
    *
+   * <p>Does nothing if the opened editor is not a text editor.
+   *
    * @param event the event to react to
    * @see FileEditorManagerListener#selectionChanged(FileEditorManagerEvent)
    */
   private void generateEditorActivatedActivity(@NotNull FileEditorManagerEvent event) {
-    localEditorHandler.activateEditor(project, event.getNewFile());
+    FileEditor newEditor = event.getNewEditor();
+
+    if (newEditor == null || newEditor instanceof TextEditor) {
```

Comment:
Have not read the API but what is the context when newEditor is  null ?

Reply:
Then no editor is selected (i.e. no editor is currently open). This information should also be transferred to other participants. This could lead to a situation where an unnecessary "no editors open" activity is send to the other participants if only non-text editors are open and the last of such editors is closed. But that should not be an issue.